hail/src/test/scala/is/hail/expr/ir/StringLengthSuite.scala

```diff
@@ -1,32 +1,19 @@
 package is.hail.expr.ir
 
-import is.hail.ExecStrategy
+import is.hail.{ExecStrategy, SparkSuite}
 import is.hail.TestUtils._
 import is.hail.expr.types._
 import is.hail.expr.types.virtual.TString
 import org.scalatest.testng.TestNGSuite
 import org.testng.annotations.Test
 
-class StringLengthSuite extends TestNGSuite {
+class StringLengthSuite extends SparkSuite {
   implicit val execStrats = ExecStrategy.javaOnly
 
   @Test def sameAsJavaStringLength() {
-    assertEvalsTo(StringLength(Str("abc")), 3)
-    assertEvalsTo(StringLength(Str("")), 0)
-  }
-
-  @Test def poopEmojiIsLengthFour() {
-    // poop is wayyy out of the basic multilingual plane (BMP), so its encoded
-    // with two UTF-16 code points. Behold, poop hex:
-    //
-    // NB: hail treats strings as arrays of 8 bit characters, so the poop emoji
-    // has length 4
-    val poopEmoji = new String(Array[Char](0xD83D, 0xDCA9))
-    assertEvalsTo(StringLength(Str(poopEmoji)), 4)
-    assertEvalsTo(StringLength(Str(poopEmoji + poopEmoji)), 8)
-  }
-
-  @Test def testIsStrict() {
-    assertEvalsTo(StringLength(NA(TString())), null)
+    val strings = Array("abc", "", new String(Array[Char](0xD83D, 0xDCA9)))
```

Comment:
I remember at some point we debated using the java functions for the truth results compared to hard coding the results. The benefit of the hard coded test was you weren't testing with the same function you were ultimately calling in the IR.

Reply:
I chose to test it this way because that's explicitly the behavior I want to test for right now---I want it to be consistent with the java string length regardless of the underlying implementation. It's true that they currently call the same underlying function, but hard coding the results (which we had before) lead to an inadvertent divergence from that.